crates/nargo/src/package.rs

```diff
@@ -2,15 +2,12 @@ use std::{collections::BTreeMap, path::PathBuf};
 
 use noirc_frontend::graph::{CrateName, CrateType};
 
-use crate::{
-    constants::{PROVER_INPUT_FILE, VERIFIER_INPUT_FILE},
-    workspace::Workspace,
-};
+use crate::constants::{PROVER_INPUT_FILE, VERIFIER_INPUT_FILE};
 
 #[derive(Clone)]
 pub enum Dependency {
-    Local { entry_path: PathBuf, crate_type: CrateType, workspace: Workspace },
-    Remote { entry_path: PathBuf, crate_type: CrateType, workspace: Workspace },
+    Local { package: Package },
+    Remote { package: Package },
```

Comment:
While working on the CrateType changes, I found that determining a entry_path and crate_type outside of a workspace didn't make sense. So this has been changed to always be resolved to a `Package`.

However, I prototyped what it would look like for us to support dependencies that contain a workspace in https://github.com/noir-lang/noir/tree/phated/workspace-as-dependencies.

The reason that I didn't introduce those changes in this PR is because it requires that the dependency name match the `name` field in the Nargo.toml and we'd need something like a `package` field to allow the dependency to be aliased (also done in that branch). This seemed like too big of a breaking change to include with the workspace changes and I think we'd want to have a separate changelog entry.

Reply:
Opened an issue at #2107